src/pydicom/values.py

```diff
@@ -60,9 +60,16 @@ def convert_tag(byte_string: bytes, is_little_endian: bool, offset: int = 0) ->
     -------
     BaseTag
         The decoded tag.
+
+    Raises
+    ------
+    TypeError
+        If `byte_string` is too short.
     """
+    if len(byte_string) < 4:
+        raise TypeError("byte string too short - must be at least 4 bytes long.")
     fmt = "<HH" if is_little_endian else ">HH"
-    value = cast(tuple[int, int], unpack(fmt, byte_string[offset : offset + 4]))
+    value = cast(tuple[int, int], unpack(fmt, byte_string[offset: offset + 4]))
```

Comment:
I think this should be ValueError, because it is a problem with the value, not with it's type. (I suspect I did this in the past too and maybe some need to be corrected)

Reply:
True, I was unhappy with TypeError, but for some reason did not think about ValueError. Will fix this tomorrow, I'm done for today...